qcodes/dataset/sqlite/queries.py

```diff
@@ -829,10 +831,41 @@ def get_guid_from_run_id(conn: ConnectionPlus, run_id: int) -> str:
     Args:
         conn: database connection
         run_id: id of the run
+
+    Returns:
+        The guid of the run_id.
     """
     return select_one_where(conn, "runs", "guid", "run_id", run_id)
 
 
+def get_guid_from_multiple_run_ids(db_path: str,
+                                   run_ids: Iterable[int]) \
+                                   -> List[str]:
+    """
+    Expanding the functionality of get_guid_from_run_id to retrive guids
+    from database and for multiple run ids.
+
+    Args:
+        db_path: The path to the database file.
+        run_ids: An integer iterable of run ids to get their guids.
+
+    Returns:
+        A list of guids belonging to the inserted run_ids.
+    """
+
+    guids: List[str] = []
+    for run_id in run_ids:
```

Comment:
```suggestion
    with connect(db_path) as conn:
        for run_id in run_ids:
            if run_exists(conn=conn, run_id=run_id):
            run_id_guid = get_guid_from_run_id(conn=conn,
                                               run_id=run_id)
            guids.append(run_id_guid)
        else:
            raise RuntimeError(f'run id {run_id} and above are not'
                               ' available in the database')
```

Reply:
ValueError                                Traceback (most recent call last)
<ipython-input-2-6a39f4a81d5d> in <module>
----> 1 get_guid_from_multiple_run_ids(db_path='doNd.db', run_ids=[1, 2, 3, 4, 5])

c:\users\a-fbonabi\qcodes\qcodes\dataset\sqlite\queries.py in get_guid_from_multiple_run_ids(db_path, run_ids)
    858     for run_id in run_ids:
    859         with connect(db_path) as conn:
--> 860             if run_exists(conn=conn, run_id=run_id):
    861                 run_id_guid = get_guid_from_run_id(conn=conn,
    862                                                     run_id=run_id)

c:\users\a-fbonabi\qcodes\qcodes\dataset\sqlite\queries.py in run_exists(conn, run_id)
   1064     );
   1065     """
-> 1066     res: sqlite3.Row = atomic_transaction(conn, query, run_id).fetchone()
   1067     return bool(res[0])
   1068 

c:\users\a-fbonabi\qcodes\qcodes\dataset\sqlite\connection.py in atomic_transaction(conn, sql, *args)
    156 
    157     """
--> 158     with atomic(conn) as atomic_conn:
    159         c = transaction(atomic_conn, sql, *args)
    160     return c

c:\users\a-fbonabi\anaconda3\envs\qcodes\lib\contextlib.py in __enter__(self)
    111         del self.args, self.kwds, self.func
    112         try:
--> 113             return next(self.gen)
    114         except StopIteration:
    115             raise RuntimeError("generator didn't yield") from None

c:\users\a-fbonabi\qcodes\qcodes\dataset\sqlite\connection.py in atomic(conn)
     82     with DelayedKeyboardInterrupt():
     83         if not isinstance(conn, ConnectionPlus):
---> 84             raise ValueError('atomic context manager only accepts '
     85                              'ConnectionPlus database connection objects.')
     86 

ValueError: atomic context manager only accepts ConnectionPlus database connection objects.

Reply:
Ok that is unfortunate. If we cannot use a context manager we will have to rely on a try finally clause to handle this. 
But in any case we should both ensure that we only open one connection and make sure that it is probably closed at the end of the function. Lets also write a test for this function so we are sure it works correctly

```
try:
   conn =connnect(dbpath)
   rest of the code here



finally:
    conn.close()
```

Reply:
Yes. We should do this.